docs/how-to/hip_runtime_api/multi_device.rst

```diff
@@ -0,0 +1,423 @@
+        HIP_CHECK(hipMemcpy(hostData0, deviceData0, size, hipMemcpyDeviceToHost));
+
+        // Copy result from device 1
+        double hostData1[1024];
+        HIP_CHECK(hipSetDevice(deviceId1));
+        HIP_CHECK(hipMemcpy(hostData1, deviceData1, size, hipMemcpyDeviceToHost));
+
+        // Display results from both devices
+        std::cout << "Device 0 data: " << hostData0[0] << std::endl;
+        std::cout << "Device 1 data: " << hostData1[0] << std::endl;
+
+        // Free device memory
+        HIP_CHECK(hipFree(deviceData0));
+        HIP_CHECK(hipFree(deviceData1));
+
+        return 0;
+    }
+
+
+Stream and event behavior
+===============================================================================
+
+In a multi-device system, streams and events are essential for efficient
+parallel computation and synchronization. Streams enable asynchronous task
+execution, allowing multiple devices to process data concurrently without
+blocking one another. Events provide a mechanism for synchronizing operations
+across devices, ensuring that tasks on one device are completed before
+dependent tasks on another device begin. This coordination prevents race
+conditions and optimizes data flow in multi-GPU systems. Together, streams and
+events maximize performance by enabling parallel execution, load balancing, and
+effective resource utilization across heterogeneous hardware.
+
+.. code-block:: cpp
+
+    #include <hip/hip_runtime.h>
+    #include <iostream>
+
+    __global__ void simpleKernel(double *data)
+    {
+        int idx = blockIdx.x * blockDim.x + threadIdx.x;
+        data[idx] = idx * 2.0;
+    }
+
+    int main()
+    {
+        int numDevices;
+        hipGetDeviceCount(&numDevices);
+
+        if (numDevices < 2) {
+            std::cerr << "This example requires at least two GPUs." << std::endl;
+            return -1;
+        }
+
+        double *deviceData0, *deviceData1;
+        size_t size = 1024 * sizeof(*deviceData0);
+
+        // Create streams and events for each device
+        hipStream_t stream0, stream1;
+        hipEvent_t startEvent0, stopEvent0, startEvent1, stopEvent1;
+
+        // Initialize device 0
+        hipSetDevice(0);
+        hipStreamCreate(&stream0);
+        hipEventCreate(&startEvent0);
+        hipEventCreate(&stopEvent0);
+        hipMalloc(&deviceData0, size);
+
+        // Initialize device 1
+        hipSetDevice(1);
+        hipStreamCreate(&stream1);
+        hipEventCreate(&startEvent1);
+        hipEventCreate(&stopEvent1);
+        hipMalloc(&deviceData1, size);
+
+        // Record the start event on device 0
+        hipSetDevice(0);
+        hipEventRecord(startEvent0, stream0);
+
+        // Launch the kernel asynchronously on device 0
+        simpleKernel<<<1000, 128, 0, stream0>>>(deviceData0);
+
+        // Record the stop event on device 0
+        hipEventRecord(stopEvent0, stream0);
+
+        // Wait for the stop event on device 0 to complete
```

Comment:
Is this a non-blocking operation? Or does the host code stop executing at this point to wait for stopEvent0? What lines in the code does this affect? Where does the host code stop to wait for the stopEvent to return?